server/src/grpc/remote_engine_service/mod.rs

```diff
@@ -162,31 +159,41 @@ impl<Q: QueryExecutor + 'static> RemoteEngineServiceImpl<Q> {
     ) -> std::result::Result<Response<WriteResponse>, Status> {
         let begin_instant = Instant::now();
         let request = request.into_inner();
-        let mut write_table_futures = Vec::with_capacity(request.batch.len());
+        let mut write_table_handles = Vec::with_capacity(request.batch.len());
         for one_request in request.batch {
             let ctx = self.handler_ctx();
-            write_table_futures.push(async move { handle_write(ctx, one_request).await });
+            let handle = self
+                .runtimes
+                .write_runtime
+                .spawn(handle_write(ctx, one_request));
+            write_table_handles.push(handle);
         }
 
-        let handle = self
-            .runtimes
-            .write_runtime
-            .spawn(try_join_all(write_table_futures));
-        let batch_result = handle.await.box_err().context(ErrWithCause {
-            code: StatusCode::Internal,
-            msg: "fail to run the join task",
-        });
-
-        let mut batch_resp = WriteResponse::default();
-        match batch_result {
-            Ok(Ok(v)) => {
-                batch_resp.header = Some(error::build_ok_header());
-                batch_resp.affected_rows = v.into_iter().map(|resp| resp.affected_rows).sum();
-            }
-            Ok(Err(e)) | Err(e) => {
-                batch_resp.header = Some(error::build_err_header(e));
-            }
+        let mut batch_resp = WriteResponse {
+            header: Some(error::build_ok_header()),
+            affected_rows: 0,
         };
+        for write_handle in write_table_handles {
+            let write_result = write_handle.await.box_err().context(ErrWithCause {
+                code: StatusCode::Internal,
+                msg: "fail to run the join task",
+            });
+            match write_result {
+                Ok(res) => match res {
+                    Ok(resp) => batch_resp.affected_rows += resp.affected_rows,
+                    Err(e) => {
+                        error!("Failed to write batches, err:{e}");
+                        batch_resp.header = Some(error::build_err_header(e));
+                        break;
+                    }
+                },
+                Err(e) => {
+                    error!("Failed to write batches, err:{e}");
```

Comment:
Make this error log different with L185?

Reply:
Doesn't make much sense, log with include line number.